src/Utility/Security.php

```diff
@@ -51,20 +51,29 @@ class Security
      * Create a hash from string using given method.
      *
      * @param string $string String to hash
-     * @param string|null $type Hashing algo to use (i.e. sha1, sha256 etc.).
+     * @param string|null $algorithm Hashing algo to use (i.e. sha1, sha256 etc.).
      *   Can be any valid algo included in list returned by hash_algos().
      *   If no value is passed the type specified by `Security::$hashType` is used.
      * @param mixed $salt If true, automatically prepends the application's salt
      *   value to $string (Security.salt).
      * @return string Hash
      * @link https://book.cakephp.org/3.0/en/core-libraries/security.html#hashing-data
      */
-    public static function hash($string, $type = null, $salt = false)
+    public static function hash($string, $algorithm = null, $salt = false)
     {
-        if (empty($type)) {
-            $type = static::$hashType;
+        if (empty($algorithm)) {
+            $algorithm = static::$hashType;
+        }
+        $algorithm = strtolower($algorithm);
+
+        $availableAlgorithms = hash_algos();
+        if (!in_array($algorithm, $availableAlgorithms)) {
```

Comment:
I always use true as in_arrays last arg unless required to not to (same as always === unless required to not to)